ksp_plugin/part_subsets.cpp

```diff
@@ -71,8 +67,7 @@ void Subset<Part>::Properties::Collect(
   collected_ = true;
   if (EqualsExistingPileUp()) {
     PileUp& pile_up = *parts_.front()->containing_pile_up();
-    pile_up.set_mass(total_mass_);
-    pile_up.set_intrinsic_force(total_intrinsic_force_);
+    pile_up.RecomputeFromParts();
```

Comment:
Aren't you making things quadratic in the number of parts instead of linear here? Part counts can get pretty big, and this runs at every frame; we should be careful.

Reply:
As discussed, this happens once when the pile-up is constructed/updated, so O(N).